src/librustc_metadata/cstore.rs

```diff
@@ -96,32 +96,30 @@ pub struct CStore {
 impl CStore {
     pub fn new(metadata_loader: Box<MetadataLoader + Sync>) -> CStore {
         CStore {
-            metas: RwLock::new(IndexVec::new()),
+            metas: RwLock::new(IndexVec::from_elem_n(None, 1)),
```

Comment:
The first entry is for `LOCAL_CRATE` here right? Do we call `set_crate_data` for that?

Reply:
Yes, that's for the local crate. The entry will always be `None`. It's just there to make array indices match `CrateNum`s.

Reply:
Could you add that as a comment?

Reply:
Done.